File: javascript/functions/file_operations.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');

const testFile = path.join(__dirname, '../tmp/test_output.txt');

const data = 'Hello from JavaScript!\nThis is a test file.\n';
fs.writeFileSync(testFile, data);
console.log(`Written to ${testFile}`);

const content = fs.readFileSync(testFile, 'utf8');
console.log(`Read content:\n${content}`);

const lines = content.split('\n').filter(line => line.length > 0);
console.log(`Lines: ${JSON.stringify(lines)}`);

if (fs.existsSync(testFile)) {
    const stats = fs.statSync(testFile);
    console.log(`File exists: ${testFile}`);
    console.log(`File size: ${stats.size} bytes`);
}

fs.appendFileSync(testFile, 'Appended line\n');
console.log(`Appended to file`);

File: go/functions/file_operations.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
)

func main() {
	testFile := "../tmp/test_output.txt"

	data := "Hello from Go!\nThis is a test file.\n"
	err := ioutil.WriteFile(testFile, []byte(data), 0644)
	if err != nil {
		fmt.Printf("Error writing file: %v\n", err)
		return
	}
	fmt.Printf("Written to %s\n", testFile)

	content, err := ioutil.ReadFile(testFile)
	if err != nil {
		fmt.Printf("Error reading file: %v\n", err)
		return
	}
	fmt.Printf("Read content:\n%s", string(content))

	info, err := os.Stat(testFile)
	if err == nil {
		fmt.Printf("File exists: %s\n", testFile)
		fmt.Printf("File size: %d bytes\n", info.Size())
	}

	file, err := os.OpenFile(testFile, os.O_APPEND|os.O_WRONLY, 0644)
	if err != nil {
		fmt.Printf("Error opening file: %v\n", err)
		return
	}
	defer file.Close()

	_, err = file.WriteString("Appended line\n")
	if err != nil {
		fmt.Printf("Error appending: %v\n", err)
	} else {
		fmt.Println("Appended to file")
	}
}
